### PR TITLE
ESD Tower system integration

### PR DESCRIPTION
**Pull request by araustin**
This pulls a report from Salesforce and does some transformations to meet a specification for the Tower system used by the landfill.

### CODE REVIEW

**Comment by araustin** [comment]
@andrelllb can we merge this in?

**Comment by andrelllb** [comment]
@araustin I looked this over -  just a few things to change. I made a few notes to myself. Should have it done within a day.

File: poseidon/dags/amcs/amcs_dags.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+dag = DAG(
+    dag_id='amcs',
+    default_args=args,
+    start_date=start_date,
+    schedule_interval=schedule
+)
+
+
+#: Latest Only Operator for amcs 
+amcs_latest_only = LatestOnlyOperator(
```
Comment (andrelllb) [inline poseidon/dags/amcs/amcs_dags.py:28]: Remove this

File: poseidon/dags/amcs/amcs_dags.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+from dags.amcs.amcs_jobs import *
+from trident.util import general
+from trident.util.notifications import notify
+from trident.util.seaboard_updates import update_seaboard_date, get_seaboard_update_dag
+
+
+args = general.args
+conf = general.config
+# use get it done until I figure out what this means
+schedule = general.schedule['get_it_done']
```
Comment (andrelllb) [inline poseidon/dags/amcs/amcs_dags.py:15]: @araustin How often do you want this to run?

File: poseidon/dags/amcs/amcs_jobs.py [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+
+previous_run_temp_file1 = conf['temp_data_dir'] + '/amcs_previous_run.csv'
+temp_file1 = conf['temp_data_dir'] + '/amcs_sites_temp.csv'
+temp_file2 = conf['temp_data_dir'] + '/cleaned_amcs_sites.csv'
+
+temp_file3 = conf['temp_data_dir'] + '/all_columns_amcs_sites.csv'
+final_file = conf['temp_data_dir'] + '/final_amcs_sites.csv'
+
+# If the server IP ever changes, the smbclient command might fail with NT_STATUS_UNSUCCESSFUL errors
+server_ip = '10.68.8.140'
```
Comment (andrelllb) [inline poseidon/dags/amcs/amcs_jobs.py:24]: Put this into environment

File: poseidon/dags/amcs/amcs_jobs.py [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+        + "--directory='Tower7Train/EPACS Import' -c '" \
+        + " put {out_f} sites_export.csv'"
+
+    command = command.format(adname=conf['svc_acct_user'],
+                             adpass=conf['svc_acct_pass'],
+                             out_f=final_file,
+                             server_ip=server_ip)
+
+    logging.info(command)
+
```
Comment (andrelllb) [inline poseidon/dags/amcs/amcs_jobs.py:41]: Include quote for security

File: poseidon/dags/amcs/amcs_jobs.py [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+    )
+
+
+    
+def get_diff(previous_file, current_file, output_file):
+    """Use diff command to find changes"""
+
+    # put the header in the output file
+    command = "head -n +1 {} > {}".format(current_file, output_file)
+    return_code = subprocess.call(command, shell=True)
```
Comment (andrelllb) [inline poseidon/dags/amcs/amcs_jobs.py:280]: Use quote

File: poseidon/dags/amcs/amcs_jobs.py [hunk 6]
```diff
@@ -0,0 +1,312 @@
...
+    logging.info(command)
+
+    if return_code != 0:
+        raise Exception('Could not add headers to output file')
+
+    # skip the header and sort the previous file
+    previous_file_sorted = '{}.sorted'.format(previous_file)
+
+    command = "tail -n +2 {} | sort > {}".format(previous_file, previous_file_sorted)
+    return_code = subprocess.call(command, shell=True)
```
Comment (andrelllb) [inline poseidon/dags/amcs/amcs_jobs.py:290]: Use quote

File: poseidon/dags/amcs/amcs_jobs.py [hunk 7]
```diff
@@ -0,0 +1,312 @@
...
+    logging.info(command)
+    if return_code != 0:
+        raise Exception('Could not sort {}'.format(previous_file))
+
+    # skip the header and sort the current file
+    current_file_sorted = '{}.sorted'.format(current_file)
+    command = "tail -n +2 {} | sort > {}".format(current_file, current_file_sorted)
+    return_code = subprocess.call(command, shell=True)
```
Comment (andrelllb) [inline poseidon/dags/amcs/amcs_jobs.py:298]: Use quote

File: poseidon/dags/amcs/amcs_jobs.py [hunk 8]
```diff
@@ -0,0 +1,312 @@
...
+    logging.info(command)
+    if return_code != 0:
+        raise Exception('Could sort {}'.format(current_file))
+
+    # diff the files and save the changes and new additions in the output file
+    command = "diff {} {} | grep '> ' | awk '{{print substr($0,3)}}' >> {}".format(previous_file_sorted, current_file_sorted, output_file)
+    logging.info(command)
+
+    return_code = subprocess.call(command, shell=True)
```
Comment (andrelllb) [inline poseidon/dags/amcs/amcs_jobs.py:307]: Use quote

**Pull request merged**